libs-scala/scala-utils/src/main/scala/com/daml/scalautil/concurrent/FutureOf.scala

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.scalautil.concurrent
+
+import scala.language.{higherKinds, implicitConversions}
+import scala.{concurrent => sc}
+import scala.util.Try
+
+import scalaz.{Catchable, Cobind, Isomorphism, Leibniz, MonadError, Nondeterminism, Semigroup}
+import Isomorphism.<~>
+import Leibniz.===
+import scalaz.std.scalaFuture._
+
+sealed abstract class FutureOf {
+
+  /** We don't use [[sc.Future]] as the upper bound because it has methods that
+    * collide with the versions we want to use, i.e. those that preserve the
+    * phantom `EC` type parameter.  By contrast, [[sc.Awaitable]] has only the
+    * `ready` and `result` methods, which are mostly useless.
+    */
+  type T[-EC, +A] <: sc.Awaitable[A]
+  private[concurrent] def subst[F[_[+ _]], EC](ff: F[sc.Future]): F[T[EC, +?]]
+}
+
+/** Instances and methods for `FutureOf`. You should not import these; instead,
+  * enable `-Xsource:2.13` and they will always be available without import.
+  */
+object FutureOf {
+  val Instance: FutureOf = new FutureOf {
+    type T[-EC, +A] = sc.Future[A]
+    override private[concurrent] def subst[F[_[+ _]], EC](ff: F[sc.Future]) = ff
+  }
+
+  type ScalazF[F[+ _]] = Nondeterminism[F]
+    with Cobind[F]
+    with MonadError[F, Throwable]
+    with Catchable[F]
+
+  implicit def `future Instance`[EC: ExecutionContext]: ScalazF[Future[EC, +?]] =
+    Instance subst [ScalazF, EC] implicitly
+
+  implicit def `future Semigroup`[A: Semigroup, EC: ExecutionContext]: Semigroup[Future[EC, A]] = {
+    type K[T[+ _]] = Semigroup[T[A]]
+    Instance subst [K, EC] implicitly
+  }
+
+  implicit def `future is any type`[A]: sc.Future[A] === Future[Any, A] =
+    Instance subst [Lambda[`t[+_]` => sc.Future[A] === t[A]], Any] Leibniz.refl
+
+  implicit def `future is any`[A](sf: sc.Future[A]): Future[Any, A] =
+    `future is any type`(sf)
```

Comment:
I'm rethinking this; @SamirTalwar-DA do you think it should be globally visible (as it is now), or should you have to import a `Conversions._` type thing, or use an explicit `asScalaUtils` method (like `JavaConverters`)?

Reply:
I think it's fine. The odds of it showing up accidentally are near-zero as far as I can see.

Reply:
My personal preference would be explicit `asScalaUtils` call. We all know that implicit `scala.collection.JavaConversions` are deprecated in favor of more explicit `scala.collection.JavaConverters`, right?

Reply:
I'm going to go with the current global availability for first iteration. We can scope it down if it seems to be a problem.